mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala

```diff
@@ -76,7 +76,17 @@ abstract class GeneralizedLinearModel(val weights: Vector, val intercept: Double
     predictPoint(testData, weights, intercept)
   }
 
-  override def toString() = "(weights=%s, intercept=%s)".format(weights, intercept)
+  /**
+   * Print a summary of the model.
+   */
+  override def toString: String = {
+    this.getClass.getName + " : " + s"intercept = $intercept, numFeatures = ${weights.size}"
+  }
+
+  /**
+   * Print the full model to a string.
+   */
+  def toDebugString: String = "(weights=%s, intercept=%s)".format(weights, intercept)
```

Comment:
... or really if someone wants to access the weights for debugging, won't they do so directly? this doesn't add much as a second `toString` method.